storage/storage_client.go

```diff
@@ -0,0 +1,118 @@
+package storage
+
+import (
+	"fmt"
+	"net/http"
+	"net/url"
+	"regexp"
+	"strings"
+	"time"
+
+	"github.com/hashicorp/go-oracle-terraform/client"
+	"github.com/hashicorp/go-oracle-terraform/opc"
+)
+
+const STR_ACCOUNT = "/Storage-%s"
+const STR_USERNAME = "/Storage-%s:%s"
+const AUTH_HEADER = "X-Auth-Token"
+const STR_QUALIFIED_NAME = "%s%s/%s"
+
+// Client represents an authenticated compute client, with compute credentials and an api client.
+type StorageClient struct {
+	client      *client.Client
+	authToken   *string
+	tokenIssued time.Time
+}
+
+func NewStorageClient(c *opc.Config) (*StorageClient, error) {
+	storageClient := &StorageClient{}
+	client, err := client.NewClient(c)
+	if err != nil {
+		return nil, err
+	}
+	storageClient.client = client
+
+	storageClient.client.APIEndpoint, err = url.Parse(fmt.Sprintf("https://%s.storage.oraclecloud.com", *client.IdentityDomain))
```

Comment:
I'd use an explicit assignment here, however. And also modify the client prior to setting the `client` field in the `storageClient` struct. 
```
endpoint, err := url.Parse(...)
if err != nil {
  ...
}
client.APIEndpoint = endpoint

storageClient.client = client
```